packages/neptune-extension/src/common/components/upload-modal/UploadForm.tsx

```diff
@@ -0,0 +1,185 @@
+}
+
+type UploadMode = 'notebook' | 'checkpoint'
+
+const UploadForm: React.FC<UploadFormProps> = ({
+  platformNotebook,
+  onClose,
+}) => {
+  const [ metadata ] = React.useState(() => platformNotebook.getMetadata());
+
+  const dispatch = useDispatch()
+  const { notebook } = useSelector(getNotebookState)
+  const { inferredUsername } = useSelector(getConfigurationState)
+
+  const noAccess = !notebook && metadata.notebookId
+  const ownerChanged = notebook && notebook.owner !== inferredUsername
+  const pathChanged = notebook && notebook.path !== metadata.path
+
+  const canUploadCheckpoint = notebook && !ownerChanged
+
+  const [ loading, setLoading ] = React.useState(false);
+
+  const [ mode, setMode ] = React.useState<UploadMode>(canUploadCheckpoint ? 'checkpoint' : 'notebook');
+
+  const [ projectId, setProjectId ] = React.useState(() => {
+    return !!notebook
+      ? notebook.projectId
+      : window.localStorage.getItem(PROJECT_LOCAL_STORAGE_KEY) || ''
+  });
+  const [ name, setName ] = React.useState('');
+  const [ description, setDescription ] = React.useState('');
+
+  function changeMode(newMode: UploadMode) {
+    if (newMode === 'checkpoint') {
+      setProjectId((notebook as NotebookDTO).projectId);
+    }
+
+    setMode(newMode);
+  }
+
+  async function handleSubmit() {
+    setLoading(true);
+
+    const content = await platformNotebook.getContent();
+
+    const checkpointMeta = {
+      path: metadata.path,
+      name,
+      description,
+    };
+
+    if (mode === 'notebook') {
+      const notebookMeta = await uploadNotebook(projectId, checkpointMeta, content);
+
+      await platformNotebook.saveNotebookId(notebookMeta.id);
+
+      const notebook = await getNotebook(notebookMeta.id)
+      dispatch(setNotebook(notebook))
+    }
+    else {
+      await uploadCheckpoint(metadata.notebookId as string, checkpointMeta, content);
+
+      if (pathChanged) {
+        // Reload notebook
+        const notebook = await getNotebook(metadata.notebookId as string)
+        dispatch(setNotebook(notebook))
+      }
+    }
+
+    setLoading(false);
+
+    onClose();
+  }
+
+  const disabled = !projectId;
+
+  return (
+    <Modal
+      isOpen
+      onClose={onClose}
+    >
+      { noAccess && (
+        <span>
+          You have no access to this notebook.
+          A copy of this notebook will be created with current checkpoint as the first one.
+        </span>
+      )}
+
+      { ownerChanged && (
+        <span>
+          You are not an owner of this notebook.
+          A copy of this notebook will be created with current checkpoint as the first one.
+        </span>
+      )}
+
+      { pathChanged && (
+        <span>
+          The path of this notebook changed.
+          A copy of this notebook will be created with current checkpoint as the first one.
+        </span>
+      )}
+
+      { canUploadCheckpoint && (
+        <React.Fragment>
+          <a href="#" onClick={() => changeMode('notebook')}>
```

Comment:
I thought these would be a radio buttons, not links or did the design changed?

Reply:
I was going to implement only the logic here and mock other things. For me it is irrelevent whether a mock is a radio or something else, mock is a mock, imagine this is just a control setting a value. I preffered a link because it is straightforward, radio control is somewhat error-prone and I did not want to spend any time on this.

Reply:
I just thought handling radio buttons is a part of the logic. Please add a task to change that to the board

Reply:
It is just a control that sets some string value. You can even replace every control with single `<input />`.